walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,202 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.HashMap;
 
 public final class FindMeetingQuery {
+
+  /**
+   * This method checks if an event contains no mandatory attendees using a MeetingRequest
+   * @param request MeetingRequest object which contains list of attendees
+   * @param event The Event object's attendees are compare with the request object
+   * @return boolean if the provide event object contains no mandatory attendees
+   */
+  private boolean eventNoMandatoryAttendees(MeetingRequest request, Event event){
+    for (String attendee : event.getAttendees()){
+      if (request.getAttendees().contains(attendee)){
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * This method checks if an event contains only optional Attendees using a MeetingRequest
+   * @param request MeetingRequest object which contains list of optional attendees
+   * @param event The Event object's attendees are compare with the request object
+   * @return boolean if the provide event object contains only optional attendees
+   */
+  private boolean optionalAttendeesEvent(MeetingRequest request, Event event){
+    for (String attendee : event.getAttendees()){
+      if (request.getAttendees().contains(attendee)){
+        return false;
+      }
+      if (!(request.getOptionalAttendees().contains(attendee))){
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * This method returns the correct time ranges if optional attendees are taken into account
+   * @param optionalEvents List of events with only optional attendees
+   * @param timeRanges A list of available TimeRanges without taken into account optional events 
+   * @return A list of TimeRanges depending on the optionalEvents list and final size of timeRanges
+   */
+  private Collection<TimeRange> resolveOptionalEventConflicts(ArrayList<Event> optionalEvents, Collection<TimeRange> timeRanges){
```

Comment:
nit: Looks like this overflows the [100 character line limit](https://google.github.io/styleguide/javaguide.html#s4.4-column-limit) set by the Java style guide. Try splitting the parameters out to their own lines.

Reply:
Ok